auth/principal.go

```diff
@@ -148,8 +148,9 @@ type User interface {
 // Used to define a user/role within DbConfig, and structures the request/response body in the admin REST API
 // for /db/_user/*
 type PrincipalConfig struct {
-	Name             *string  `json:"name,omitempty"`
-	ExplicitChannels base.Set `json:"admin_channels,omitempty"`
+	Name             *string                                       `json:"name,omitempty"`
+	ExplicitChannels base.Set                                      `json:"admin_channels,omitempty"`
+	CollectionAccess map[string]map[string]*CollectionAccessConfig `json:"collection_access"`
```

Comment:
Should we add `omitempty` to reduce storage overhead for non-collection deployments?

```suggestion
	CollectionAccess map[string]map[string]*CollectionAccessConfig `json:"collection_access,omitempty"`
```